android/WorkshopCompleted/app/src/main/java/com/mobiledgex/workshopskeleton/MainActivity.java

```diff
@@ -89,10 +89,18 @@
     private Activity ctx;
     private String host;
     private int port;
-    private String carrierName;
-    private String appName;
-    private String devName;
-    private String appVersion;
+
+    // NOTICE: In a real app, these values would be determined by the SDK, but we are reusing
+    // an existing app so we don't have to create new app provisioning data for this workshop.
+    private String carrierName = "wifi";
+    private String appName = "MobiledgeX SDK Demo";
+    private String devName = "MobiledgeX";
+    private String appVersion = "2.0";
```

Comment:
Is there a reason all of these variables are duplicated in MainActivity and FaceProcessorActivity? Can't they be in just one place?

Reply:
I will look into not duplicating these variables.